docs/_en/dev/feature-technical-designs/1696-gcp-secrets.md

```diff
@@ -0,0 +1,277 @@
+| `tock_mongodb_credentials_secret_name`| `PROD-TOCK-MONGODB` | GCP Secret name Path | Secret ID use for Mongo DB secret. Only if not passed in mongo URI using the `tock_mongo_url`. You should also include the GCP mongodb secret module to use it. |
+| `tock_gcp_secret_project_number`| `478160847739` | GCP project number | The GCP project number where secrets are stored. |
+
+### Bot API
+
+|Environment variable name | Default | Allowed values | Description |
+|--- |--- |--- |--- |
+| `tock_database_credentials_provider`| `Raw` | `Raw`, `AwsSecretsManager`, `GcpSecretManager` | Type of credential provider used to store database (mongodb) access secrets : <br> <ul> <li>Raw: Store secret directly into tock's mongodb in raw. Use it only for local dev purposed it's clearly unsafe.</li><li>AwsSecretsManager: rely on AWS Secret Manager.</li><li>GcpSecretManager: rely on GCP Secret Manager.</li></ul> |
+| `tock_iadvize_credentials_provider`| `Raw` | `Raw`, `AwsSecretsManager`, `GcpSecretManager` | Type of credential provider used to store iAdvize secrets : <br> <ul> <li>Raw: Store secret directly into tock's mongodb in raw. Use it only for local dev purposed it's clearly unsafe.</li><li>AwsSecretsManager: rely on AWS Secret Manager.</li><li>GcpSecretManager: rely on GCP Secret Manager.</li></ul> |
+| `tock_gen-ai_credentials_provider`| `Raw` | `Raw`, `AwsSecretsManager`, `GcpSecretManager` | Type of credential provider used to store generative AI secrets : <br> <ul> <li>Raw: Store secret directly into tock's mongodb in raw. Use it only for local dev purposed it's clearly unsafe.</li><li>AwsSecretsManager: rely on AWS Secret Manager.</li><li>GcpSecretManager: rely on GCP Secret Manager.</li></ul> |
+| `tock_mongodb_credentials_secret_name`| `PROD-TOCK-MONGODB` | GCP Secret name Path | Secret name use for Mongo DB secret. Only if not passed in mongo URI using the `tock_mongo_url`. You should also include the GCP mongodb secret module to use it. |
+| `tock_gcp_secret_project_number`| `478160847739` | GCP project number | The GCP project number where secrets are stored. |
+| `tock_gcp_iadvize_credentials_secret_name`| `PROD-TOCK-IADVIZE` | GCP Secret ID Path | Secret name use for Mongo DB secret iAdvize GraphQL calls. Only needed if you use iAdvize, it can also be passed though environment variables `iadvize_username_authentication` and `iadvize_password_authentication`. | 
+
+### Gen AI Orchestrator
+
+|Environment variable name | Default | Allowed values | Description |
+|--- |--- |--- |--- |
+| `tock_gcp_secret_project_number`| `478160847739` | GCP project number | The GCP project number where secrets are stored. |
+
+
+## Technical change that should be made
+
+### Breaking changes
+
+* Default value of `tock_gen_ai_orchestrator_secret_storage_prefix_name` currently `/dev` shouldn't use slashes as it's not allow according to GCP Secret Names constraints. It will be changed to `dev` but it might break all running project that doesn't defined it.
```

Comment:
The ‘break’ is not linked to the presence of the slash, as the prefix for GCP is also standardised. However, it is linked to the default value of the prefix, which has been changed to ‘LOCAL/TOCK’.

The following is a list of the changes made to the existing environment variables: 
'tock_database_credentials_provider' replaced by 'tock_database_mongodb_secret_manager_provider'
'tock_iadvize_credentials_provider' replaced by 'tock_iadvize_secret_manager_provider'
'aws_iadvize_credentials_secret_id' replaced by 'tock_iadvize_credentials_secret_name'
'tock_gen_ai_orchestrator_secret_storage_type' replaced by 'tock_gen_ai_secret_manager_provider'
'tock_gen_ai_orchestrator_secret_storage_prefix_name' replaced by 'tock_gen_ai_secret_prefix'

Reply:
Added at the end for @AurelienSylvan to be renamed / updated in helm charts also.
@sacquatella those environment variables will have impact on the Helm Charts they are all specified in this design document that will be available on the EN version of tock website, I'll create on issue on https://github.com/theopenconversationkit/tock-helm-chart so that we don't forget it.

Reply:
Issue created here : https://github.com/theopenconversationkit/tock-helm-chart/issues/2